### PR TITLE
case dev-2662: removing double calls to switch devices, handling default changes fro…

### PR DESCRIPTION
**Pull request by amerhifi**
https://highfidelity.atlassian.net/browse/DEV-2662

Moved default device switching to scripting interface. When devices list is switched the scripting interface will automatically fire a call to change device which was causing double calls if defaults changed. Moved logic to determine if defaults have changed to the scripting interface.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Can someone from High Fidelity approve this PR?

File: interface/src/scripting/AudioDevices.cpp [hunk 1]
```diff
@@ -367,19 +364,6 @@ void AudioDeviceList::onDevicesChanged(QAudio::Mode mode, const QList<HifiAudioD
                 } else {
                     desktopIsSelected = isSelected;
                 }
-
```
Comment (amerhifi) [inline interface/src/scripting/AudioDevices.cpp]: the old behavior to go back to an old plugged device is no longer valid we go do defaults devices only.

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22311/HighFidelity-Beta-PR16449-b10f2a8.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22311/HighFidelity-Quest-Beta-PR16449-b10f2a8.apk)

**Comment by hd23408** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/22317/HighFidelity-Beta-PR16449-b10f2a8.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/22317/HighFidelity-Beta-Interface-PR16449-b10f2a8.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22317/HighFidelity-Beta-PR16449-b10f2a8.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22317/HighFidelity-Quest-Beta-PR16449-b10f2a8.apk)

**Comment by hd23408** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/22321/HighFidelity-Beta-PR16449-b10f2a8.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/22321/HighFidelity-Beta-Interface-PR16449-b10f2a8.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22321/HighFidelity-Beta-PR16449-b10f2a8.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22321/HighFidelity-Quest-Beta-PR16449-b10f2a8.apk)

**Comment by JeffClinton** [comment]
Build this please

**Comment by hd23408** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22341/HighFidelity-Beta-PR16449-b10f2a8.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22341/HighFidelity-Quest-Beta-PR16449-b10f2a8.apk)

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22347/HighFidelity-Beta-PR16449-bc41087.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22347/HighFidelity-Quest-Beta-PR16449-bc41087.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/22347/HighFidelity-Beta-PR16449-bc41087.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/22347/HighFidelity-Beta-Interface-PR16449-bc41087.exe

**Comment by amerhifi** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22354/HighFidelity-Beta-PR16449-bc41087.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22354/HighFidelity-Quest-Beta-PR16449-bc41087.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/22354/HighFidelity-Beta-PR16449-bc41087.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/22354/HighFidelity-Beta-Interface-PR16449-bc41087.dmg

**Comment by hd23408** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/22360/HighFidelity-Beta-PR16449-bc41087.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/22360/HighFidelity-Beta-Interface-PR16449-bc41087.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/22360/HighFidelity-Beta-PR16449-bc41087.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/22360/HighFidelity-Beta-Interface-PR16449-bc41087.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22360/HighFidelity-Beta-PR16449-bc41087.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22360/HighFidelity-Quest-Beta-PR16449-bc41087.apk)

**Pull request merged**